### ISSUE TITLE
Allow vue/no-dupe-keys rule to accept destructured props in the return function of setup

### ISSUE THREAD

**Issue by enochndika** [vuejs/eslint-plugin-vue#1526]
**My Environnementt**

- **ESLint version:**  7.29.0
- **eslint-plugin-vue version:** 7.12.1
- **Node version:**  14.17.1

i am using destructred props with `toRefs` to keep up the reactivity but when i return that prop, eslint triggers the `Duplicated key 'author'.(vue/no-dupe-keys` error .  why the destructured prop is marked as duplicated?

```vue
<template>
  <h1>{{ author }}</h1>
</template>

<script>
import { toRefs } from "vue";
export default {
  props: {
    author: String,
  },
  setup(props) {
    const { author } = toRefs(props);

    return { author };
  },
};
</script>
```

The workaround is to turn off this rule but it's not a better solution.

**Comment by ota-meshi** [comment]
I think your source code needs to be modified as follows:

```vue
<template>
  <h1>{{ author }}</h1>
</template>

<script>
import { toRefs } from "vue";
export default {
  props: {
    author: String,
  },
  setup(props) {
    const { author } = toRefs(props);

    return { }; // <-
  },
};
</script>
```

**Comment by enochndika** [comment]
With this solution, I will end up with the error  `'author' is assigned a value but never used.(no-unused-vars)`. and because I run eslint on pre-commt hook commit, then this will fail. for the moment I will have to disable this rule to avoid all this

**Comment by LinusBorg** [comment]
> Why the destructured prop is marked as duplicated?

Because `author` is already available as a property because it's a prop of the component. overriding a prop this way is not a good idea and so the rule warns you about it.

**Comment by enochndika** [comment]
So how could i fix this ?
@LinusBorg If I don't pass  props as a params in setup, will props still be reactive?

```vue
<template>
  <h1>{{ author }}</h1>
</template>

<script>
export default {
  props: {
    author: String,
  },
  setup() {},
};
</script>

```

**Comment by LinusBorg** [comment]
Yes
